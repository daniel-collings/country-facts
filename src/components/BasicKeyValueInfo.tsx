import React from 'react'
interface IBasicKeyValueInfoProps {
  label: string
  value: string | number
}

export default function BasicKeyValueInfo({
  label,
  value
}: IBasicKeyValueInfoProps) {
  return (
    <div className="p-4 grid">
      <span className="font-bold">{label}</span>
      <span>{value}</span>
    </div>
  )
}
