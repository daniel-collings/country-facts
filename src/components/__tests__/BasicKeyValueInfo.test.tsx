import { describe, expect, test } from 'vitest'
import { render } from '@/utils/testUtils.tsx'
import BasicKeyValueInfo from '@/components/BasicKeyValueInfo.tsx'

describe('BasicKeyValueInfo Component', () => {
  test('should render correctly with string value', () => {
    const { asFragment } = render(
      <BasicKeyValueInfo value="Â£ (Great British Pound)" label="Currency" />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
