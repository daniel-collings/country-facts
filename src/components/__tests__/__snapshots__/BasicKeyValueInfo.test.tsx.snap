// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BasicKeyValueInfo Component > should render correctly with string value 1`] = `
<DocumentFragment>
  <div
    class="h-full min-h-screen bg-base-100 p-4 sm:p-8"
    data-theme="light"
  >
    <main
      class="max-w-7xl mx-auto px-0 sm:px-6 lg:px-8"
    >
      <div
        class="p-4 grid"
      >
        <span
          class="font-bold"
        >
          Currency
        </span>
        <span>
          Â£ (Great British Pound)
        </span>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
